pkg/kubectl/describe.go

```diff
@@ -1801,7 +1833,7 @@ func describeServiceAccount(serviceAccount *api.ServiceAccount, tokens []api.Sec
 			tokenSecretNames = []string{}
 		)
 
-		for _, s := range serviceAccount.ImagePullSecrets {
+		for _, s := range saPullSecrets {
```

Comment:
I meant that it seems like describe should show the service account as it actually is. if it is referencing secrets which don't exist, I'd expect to see the secret names when describing the service account. if we want to do additional work so we can indicate which secrets exist and which don't, that may be ok, but don't hide the information here.